torch/testing/_internal/common_methods_invocations.py

```diff
@@ -3635,17 +3635,23 @@ def sample_inputs_to_sparse(op_info, device, dtype, requires_grad, **kwargs):
             SampleInput(make_arg((S, S)), args=(1,), output_process_fn_grad=lambda x: x.to_dense()),)
 
 
-def sample_inputs_log_softmax(op_info, device, dtype, requires_grad, with_dtype=False, **kwargs):
+# Used for both log_softmax and softmax
+def sample_inputs_softmax_variant(op_info, device, dtype, requires_grad, with_dtype=False, **kwargs):
     make_arg = partial(make_tensor, device=device, dtype=dtype, requires_grad=requires_grad)
 
-    if with_dtype:
-        cases = (((S, S, S), (1, ), {'dtype': torch.float64}),)
-    else:
-        cases = (((S, S, S), (1,), {}),)  # type:ignore[assignment]
+    cases = (
+        ((S, ), (0, )),
+        ((S, S), (0, )),
+        ((S, S), (1, )),
+        ((S, M, S), (2, ))
+    )
 
     def generator():
-        for shape, args, kwargs in cases:
-            yield SampleInput(make_arg(shape), args=args, kwargs=kwargs)
+        for shape, dim in cases:
+            if with_dtype:
+                yield SampleInput(make_arg(shape), args=dim, kwargs={'dtype': torch.float64})
+            else:
+                yield SampleInput(make_arg(shape), args=dim)
```

Comment:
Since we are touching this anyway, I wonder if we should keep the weird generator syntax here? I mean why create a generator if the only thing we do with it is to covert it to a list?

```suggestion
return [
    SampleInput(make_arg(shape), args=dim, kwargs=dict(dtype=torch.float64) if with_dtype else None)
    for shape, dim in cases
]
```

Reply:
Thanks for the suggestion, @pmeier! This looks cleaner.